experimental/emulated_vault/opt/emulated-vault/vault-server

```diff
@@ -0,0 +1,249 @@
+            self.end_headers()
+            self.wfile.write('{"Failure"}')
+            return
+
+        self.send_response(204)
+        self.send_header('Content-Type', 'application/json')
+        self.end_headers()
+        return
+
+
+    def _do_DELETE(self):
+        logger.info("DELETE %s", self.path)
+        parsed_path = urlparse.urlparse(self.path)
+
+        success = adaptor.deleteParameter(parsed_path.path[1:])
+        if not success:
+            self.send_response(503)
+            self.send_header('Content-Type', 'application/json')
+            self.end_headers()
+            self.wfile.write('{"Failure"}')
+            return
+
+        self.send_response(200)
+        self.send_header('Content-Type', 'application/json')
+        self.end_headers()
+        self.wfile.write('{"Deleted"}')
+        return
+
+
+
+if __name__ == '__main__':
+    progPath = inspect.stack()[-1][1]
+    progAbsPath = os.path.abspath( progPath )
+    progAbsPath = os.path.dirname(os.path.normpath(progAbsPath))
+
+    confDir = os.path.join(progAbsPath, "conf")
+    confFile = os.path.join(confDir, "cfg.json")
+    with open(confFile) as fd:
+        cfg = json.load(fd)
+
+    logDir = os.path.join(progAbsPath, "var/log")
+    debugLogFile = os.path.join(logDir, "traffic-ops-vault-debug.log")
+    mainLogFile = os.path.join(logDir, "traffic-ops-vault.log")
+    if not os.path.exists(logDir):
+        os.makedirs(logDir)
+
+    global logger
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.INFO)
+    # create file handler which logs even debug messages
+    fhd = logging.FileHandler(debugLogFile)
+    fhd.setLevel(logging.DEBUG)
+    fhm = logging.FileHandler(mainLogFile)
+    fhm.setLevel(logging.INFO)
+    # create formatter and add it to the handlers
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    fhm.setFormatter(formatter)
+    fhd.setFormatter(formatter)
+    # add the handlers to the logger
+    logger.addHandler(fhm)
+    logger.addHandler(fhd)
+
+    basePath = cfg["data"]["db-path"]
+    pingRelPath = cfg["data"]["ping-rel-path"]
+    listenIP = cfg["data"]["listen-ip"]
+    listenPort = cfg["data"]["listen-port"]
+    adaptorType = cfg["data"]["adaptor-type"]
+    use_ssl = cfg["data"]["use-ssl"]
+
+    global adaptor
+    if adaptorType == "fs":
+        adaptor = adaptor.fs.Fs(logger=logger, basePath=basePath, pingRelPath=pingRelPath)
+    else:
+        logger.error("Invalid adaptor type '%s'", adaptorType)
+        sys.exit(1)
+
+    if not adaptor.init_ping():
+        logger.error("Failed initialization")
+        sys.exit(1)        
+
+    server = HTTPServer((listenIP, listenPort), RequestHandler)
+    if use_ssl:
+        key = cfg["data"]["ssl-key-path"]
+        cert = cfg["data"]["ssl-cert-path"]
+        server.socket = ssl.wrap_socket (server.socket, keyfile=key, certfile=cert, server_side=True)
```

Comment:
two unhandled `OSError`s (probably `FileNotFoundError` or `PermissionDeniedError`) and an `ssl.SSLError` here.